### DISCUSSION TITLE
Does Tag invalidation work with queryFn?

### DISCUSSION THREAD

**Discussion by ghasemikasra39** [reduxjs/redux-toolkit#1413] [Q&A] [answered]
I have two endponts, `getScrollingNote` and `createComment`. I want `getScrollingNote` triggers whenever I use `createComment`: 

```javascript
export const scrollingNoteApi = createApi({
  reducerPath: 'scrollingNote',
  baseQuery: fetchBaseQuery({ ... }),
  tagTypes: ['Comment'],
  endpoints: builder => ({
    getScrollingNote: builder.query(
        {query: getScrollingNoteHandler, providesTags: ['Comment']}),
    createComment: builder.mutation(
        {queryFn: createCommentHandler, invalidatesTags: ['Comment']})
  })
})
```

To have a cleaner code, I have these handlers:

```javascript
const getScrollingNoteHandler = notesList => { ... }
const createCommentHandler = async (args, queryApi, extraOptions, baseQuery) => { ... }
```

I have a functional component that subscribes to `getScrollingNote` and uses `createComment` via `useGetScrollingNoteQuery` and `useCreateCommentMutation`:

```javascript
function myFunction() {
  const {data} = useGetScrollingNoteQuery(someInputThatINeedToPass)
  const [createComment] = useCreateCommentMutation()
}
```

But when I use `createComment({ ... })`, my mutation goes through successfully, but the `getScrollingNote` does not trigger ! (The desired behaviour is: It must trigger)

I cannot find my mistake, am I making a mistake in `providesTags` and `invalidatesTags`?

**Comment by phryneas** [comment, marked as answer]
That would work. Did you maybe forget to set up the middleware?

**Reply by ghasemikasra39** [reply]
I cannot believe that I completely forgot that :) I added it and works perfectly. Thank you, you saved my day
